src/goalServices/helpers.js

```diff
@@ -22,7 +22,12 @@ const findOrFailExistingGoal = (needle, haystack, translate = goalFieldTransate)
     && g[translate.name].trim() === needle.name.trim()
     && g[translate.source] === needle.source
     && g[translate.responsesForComparison] === responsesForComparison(needle)
-    && haystackCollaborators.some((c) => needleCollaborators.includes(c))
+    && (
+      // Check if both needle and haystack goal have no valid collaborators
+      (needleCollaborators.length === 0 && (g.collaborators || [])
+        .every((c) => c.goalCreatorName === undefined))
+      || haystackCollaborators.some((c) => needleCollaborators.includes(c))
```

Comment:
LGTM. Might be nice to see a test explicitly assert what this is doing - I find all the inline comparisons confuse me when I return to them.